Automattic/jetpack
# Pay with PayPal block: Handle AMP requests

Follows up https://github.com/Automattic/jetpack/pull/17256.

#### Changes proposed in this Pull Request:
Since the PayPal button does not work on AMP requests, this PR ensures the output of the "Pay with PayPal" block displays a link to the non-AMP URL when the post is visited through AMP.

Before | After
--- | ---
<img width="657" alt="Screen Shot 2020-09-29 at 12 55 49" src="https://user-images.githubusercontent.com/1233880/94552705-95965600-0257-11eb-9131-da51e3b26983.png"> | <img width="531" alt="Screen Shot 2020-09-30 at 15 48 09" src="https://user-images.githubusercontent.com/1233880/94693747-5ee23d80-0334-11eb-9a5c-fe8b8d2fa3e1.png">

Since this block can be used for more things than purchases (e.g. donations), this PR also adds the ability to define a custom text for the link that is used as fallback in other contexts than frontend.

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- [Spin up a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branch=/update/pay-with-paypal-amp).
- Set up Jetpack and purchase any paid plan.
- Go to WP Admin > Plugins > Add new.
- Install and activate the [AMP plugin](https://wordpress.org/plugins/amp/).
- Go to WP Admin > Posts > Add new.
- Insert a Pay with PayPal block.
- Open the sidebar.
- Change the purchase link text.
- Publish the post.
- Visit the AMP version of the published post by going to the post URL with a `?amp` added at the end (e.g.: `https://example.com/2020/09/29/post-slug/?amp`).
- Make sure the Pay with PayPal block displays a textual call-to-action with the text you entered above that links to the non-AMP version of the post.
- Go to WP Admin > AMP.
- Switch to the "Standard" mode.
- Visit the AMP version of the published post.
- Make sure the non-AMP version is displayed (the one with a PayPal button).

#### Proposed changelog entry for your changes:
Pay with PayPal block: Handle AMP requests
